### PR TITLE
fix failover if no files or main image

### PR DESCRIPTION
**Pull request by matthew1809**
Add if else logic within the `isThereAFile` function to stop it dying if there's no file object in the relationships for products.

### CODE REVIEW

File: src/components/Products/ProductImage.js [hunk 1]
```diff
@@ -19,12 +19,24 @@ class ProductImage extends Component {
...
     
     var isThereAFile = (product) => {
-      fileId = this.props.product.relationships.files.data[0].id;
-      file = this.props.products.included.files.find(function (el) {
-        return fileId === el.id
-      })      
-        return <img alt={this.props.product.name + ', ' + this.props.product.description} src={file.link.href} style={{"background": this.props.background}}/> || <img alt="placeholder" src={placeholder}/>
-    };
+      
+      if(this.props.product.relationships.files) {
```
Comment (ynnoj) [inline src/components/Products/ProductImage.js:23]: We should use the `product` object passed into `isThereAFile()` as an argument, or remove the parameter on line 44.

Comment (ynnoj) [reply src/components/Products/ProductImage.js:23]: ☝️ Same applies to the `isThereAMainImage()` function.

File: src/components/Products/ProductImage.js [hunk 2]
```diff
@@ -19,12 +19,24 @@ class ProductImage extends Component {
...
+        
+        fileId = this.props.product.relationships.files.data[0].id;
+        
+        file = this.props.products.included.files.find(function (el) {
+          return fileId === el.id
+        })    
+          
+          return <img alt={this.props.product.name + ', ' + this.props.product.description} src={file.link.href} style={{"background": this.props.background}}/>
```
Comment (ynnoj) [inline src/components/Products/ProductImage.js:31]: I can see [instances](https://github.com/moltin/react-demo-store/blob/2b25d145ef472d6effa22e2af84e4153a891d5ca/src/components/Products/AllProducts.js#L36) where the `ProductImage` component is invoked without a `background` property. What happens in those cases?

**Review by ynnoj — CHANGES_REQUESTED** [review]

**Comment by matthew1809** [comment]
Closing in favour on changes on branch https://github.com/moltin/react-demo-store/tree/fix%2Fgracefully-handle-responses

**Pull request closed without merging**